regression-test-harness/proposals/regression_test_harness_upgrades.md

```diff
@@ -0,0 +1,152 @@
+to investigate the causes of a failed test.
+
+This naming scheme will be preserved and extended.  The `var/lib` prefix will
+be removed.  In its place will be a parent directory named after the name of
+the test being run as it is known to `ctest`.  This allows all tests to run
+simultaneously while maintaining the required filesystem isolation without
+the loss of human navigability that would be incurred by randomly generated
+temporary directory names.
+
+In addition, the path names for a cluster and for individual nodes will be
+customizable to facilitate cases where greater clarity can be achieved by
+specific node directory names.
+
+### Network usage
+
+A running `nodeos` instance listens on up to four different network ports,
+depending on which plugins are enabled, and there is a community requested
+feature which would expand that to five.  Berkley sockets restrict listening
+to one process per IP address and port, requiring coordination between nodes
+in a cluster and across clusters to avoid collisions.
+
+The current test harness hardcodes an initial listen port of 8888 on 127.0.0.1
+and uses each port in sequence per node in the cluster for the http plugin.
+The revised system will take advantage of the entirety of the 127.0.0.0/8
+subnet by choosing an IP address to run the cluster on deterministically based
+on the name of the test as it is known to `ctest`.  The listen port for the
+http plugin will remain the default 8888 and sequential ports, though at least
+one test will start on an alternate port to ensure customized ports work
+throughout the system.
+
+As with filesystem usage, the IP address and ports used will be customizable.
+The test harness currently exposes the port number for the HTTP plugin for
+customization but does not honor it in all cases.  This defect will be
+corrected and the option for customization extended to the other
+network-enabled plugins.  The IP address will be exposed for customization.
+The `eosio-launcher` currently supports custom IP addresses, including remote
+IP addresses.  This functionality will be preserved while reimplementing it
+in Python.
+
+## Improved support for typical scenarios
+
+The test harness serves dual purposes: it executes as part of the automated
+testing invoked prior to accepting a Github pull request and it is used by
+developers on their own machines to quickly and easily compose test scenarios
+during development.
+
+### Improved CI/CD support
+
+To improve the utility of the test harness in CI/CD scenarios, it will:
+
+- Provide support for executing tests in mixed network scenarios, including:
+  - Release builds from multiple versions;
+  - Debug builds;
+  - Pinned builds.
+- Execute code coverage tests in cooperation with CMake.
+- Capture logs and other test artifacts into archives on commmand for upload to:
+  - Github test results;
+  - A status dashboard showing test results and code coverage statistics.
```

Comment:
How/where will the dashboard be created?

Reply:
Ted requested it.  Whether or not its data source is a Github action and where it would be hosted is yet to be determined.